### PR TITLE
Add video files and datepicker field

### PR DESCRIPTION
**Pull request by davidmiller11**
Why?
----
* Allow users to upload video files.
* Allow users to include a datepicker field.

How?
----
* Add `AtomicCms::Video` model
* Add `_datepicker_field` template
* Add `http://s3.amazonaws.com/**` to Angular whitelisted domains
* Add errors to json response in `Atomic::Cms::MediaController`
* Display parsed errors on `_file_field` template
* Bump version to 0.2.6
* Add specs

### CODE REVIEW

**Comment by samcdavid** [comment]
FWIW, I'm pretty sure there is a checklist somewhere for publishing the gem once this gets merged.

**Comment by samcdavid** [comment]
also `rubocop -a` kthxbai

**Comment by davidmiller11** [comment]
Fixed the hound issues.  @samcdavid This gem-publishing checklist you speak of...its an internal checklist?

**Comment by samcdavid** [comment]
@davidmiller11 yea, I think it is in a spartan repo, like spartansystems/spartan or something like that.

**Comment by gabeweaver** [comment]
@davidmiller11 it's a [super lengthy process](https://github.com/spartansystems/spartan/blob/master/docs/Delivery/Open_Source_Release_Checklist.md). let me know if you get stuck :trollface:

**Comment by davidmiller11** [comment]
## Ruby Open Source Gem Release Checklist

### Pre Merge into Master

- [ ] Bump version numbers and dates in `gemspec` following [semver](http://semver.org/).

### Post Merge

- [ ] Checkout the master branch and pull the latest changes.
- [ ] Publish the gem with `rake publish`

**Comment by squaresurf** [comment]
From what I remember, I think that bundler has a rake task that can be added that can automate the versioning, tagging and all as well as pushing to ruby gems. Might be something to look at for the future.

File: .rubocop.yml [hunk 1]
```diff
@@ -1,5 +1,33 @@
...
+Style/FrozenStringLiteralComment:
+  EnforcedStyle: when_needed
+  SupportedStyles:
+    - when_needed
+    - always
+    - never
+
 Style/StringLiterals:
-  Enabled: false
+  EnforcedStyle: single_quotes
```
Comment (samcdavid) [inline .rubocop.yml]: I don't like this, but I literally couldn't give a flying 💩  for it anymore.

File: atomic_cms.gemspec [hunk 2]
```diff
@@ -1,14 +1,14 @@
 Gem::Specification.new do |s|
   s.name        = 'atomic_cms'
-  s.version     = '0.2.5'
+  s.version     = '0.3.0'
```
Comment (samcdavid) [inline atomic_cms.gemspec]: :smile:

File: app/controllers/concerns/media_scrubber.rb [hunk 3]
```diff
@@ -8,12 +8,16 @@ def initialize(args)
...
+    return AtomicCms::Image.new(params) if original.content_type =~ /image/
+    AtomicCms::Video.new(params) if original.content_type =~ /video/
   end
 
   def valid?
     return false unless filtered
-    filtered.valid?
+    return true if filtered.valid?
+    @errors = filtered.errors
+    false
```
Comment (TheAznShumai) [inline app/controllers/concerns/media_scrubber.rb]: Not super important, but I think we can save a line by doing this.
```
return true if filtered&.valid?
@errors = filtered&.errors
false
```

Comment (davidmiller11) [inline app/controllers/concerns/media_scrubber.rb]: @TheAznShumai too late :)

**Pull request merged**